roles/zabbix_agent/tasks/main.yml

```diff
@@ -43,7 +43,7 @@
     zabbix_agent_heartbeatfrequency: "{{ zabbix_agent_heartbeatfrequency is defined | ternary(zabbix_agent_heartbeatfrequency, zabbix_agent2_heartbeatfrequency) | default(60) }}"
     zabbix_agent_hostname: "{{ zabbix_agent_hostname is defined | ternary(zabbix_agent_hostname, zabbix_agent2_hostname) | default(inventory_hostname) }}"
     zabbix_agent_include_mode: "{{ zabbix_agent_include_mode is defined | ternary(zabbix_agent_include_mode, zabbix_agent2_include_mode) | default('0750') }}"
-    zabbix_agent_listenport: "{{ zabbix_agent_listenport is defined | ternary(zabbix_agent_listenport, zabbix_agent2_listenport) | default(10050) }}"
+    zabbix_agent_listenport: "{{ zabbix_agent_listenport | default(zabbix_agent2_listenport) | default(10050) }}"
```

Comment:
Sorry, I don't get this change. Why is it better than `ternary`?

Reply:
Because the tenary does not work when `zabbix_agent2_listenport` is not defined.
I have tested it.

Reply:
Its works fine in all of the lines above it, not sure why it would be different on this one.  Are you sure its not a matter of `zabbix_agent2_listenport` being null instead of not defined?

Reply:
OK you are right. We don't need this change because it is already fixed by this change https://github.com/ansible-collections/community.zabbix/pull/1259/files#diff-d722eb1d2f161071c11093c3875382316f4cc4b6bf5d37be35bbb3b5ac2e283aR91